// react-bootstrap-storybook/dist/Form/Form.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Form = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Form/Form',
  component: _index.Form,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Form, null, /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3",
    controlId: "formBasicEmail"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "Email address"), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "email",
    placeholder: "Enter email"
  }), /*#__PURE__*/_react.default.createElement(_index.Form.Text, {
    className: "text-muted"
  }, "We'll never share your email with anyone else.")), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3",
    controlId: "formBasicPassword"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "Password"), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "password",
    placeholder: "Password"
  })), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3",
    controlId: "formBasicCheckbox"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
    type: "checkbox",
    label: "Check me out"
  })));
};

var _Form = Template.bind({});

exports._Form = _Form;
_Form.args = {};
