// react-bootstrap-storybook/dist/Form/Form.FloatingLabel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Floatinglabel = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Form/Floatinglabel',
  component: _index.Form,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Form, null, /*#__PURE__*/_react.default.createElement("h3", null, "Form.FloatingLabel"), /*#__PURE__*/_react.default.createElement(_index.Form.FloatingLabel, {
    controlId: "floatingInput",
    label: "Email address",
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "email",
    placeholder: "name@example.com"
  })), /*#__PURE__*/_react.default.createElement(_index.Form.FloatingLabel, {
    controlId: "floatingPassword",
    label: "Password"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "password",
    placeholder: "Password"
  })));
};

var Floatinglabel = Template.bind({});
exports.Floatinglabel = Floatinglabel;
Floatinglabel.args = {};
