// react-bootstrap-storybook/dist/Form/Form.Control.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Control = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Form/Control',
  component: _index.Form,
  argTypes: {}
};
exports.default = _default;
var types = [// 'button',
// 'checkbox',
'color', 'date', 'datetime-local', 'email', 'file', 'hidden', // 'image',
'month', 'number', 'password', // 'radio',
// 'range',
// 'reset',
'search', // 'submit',
// 'tel',
'text', 'time' // 'url',
// 'week'
];

var FormControlTemplate = function FormControlTemplate(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Form, null, /*#__PURE__*/_react.default.createElement("h3", null, "Form.Control type"), types.map(function (item) {
    return /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
      key: item,
      className: "mb-3"
    }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "type=", item), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
      type: item,
      placeholder: "input ".concat(item)
    }));
  }), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "as=\"textarea\""), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    as: "textarea",
    rows: 3
  })), /*#__PURE__*/_react.default.createElement("h3", null, "Form.Control size"), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "size=\"lg\""), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    size: "lg",
    type: "text",
    placeholder: "Large text"
  })), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "size=\"\""), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "text",
    placeholder: "Normal text"
  })), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "size=\"sm\""), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    size: "sm",
    type: "text",
    placeholder: "Small text"
  })), /*#__PURE__*/_react.default.createElement("h3", null, "Form.Control readOnly"), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "text",
    placeholder: "Readonly input here...",
    readOnly: true
  })), /*#__PURE__*/_react.default.createElement("h3", null, "Form.Control plaintext"), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    plaintext: true,
    defaultValue: "email@example.com"
  })), /*#__PURE__*/_react.default.createElement("h3", null, "Form.Control File input"), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "multiple=false"), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "file"
  })), /*#__PURE__*/_react.default.createElement(_index.Form.Group, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.Form.Label, null, "multiple"), /*#__PURE__*/_react.default.createElement(_index.Form.Control, {
    type: "file",
    multiple: true
  })));
};

var Control = FormControlTemplate.bind({});
exports.Control = Control;
Control.args = {};
