// react-bootstrap-storybook/dist/Form/Form.Select.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Form/Select',
  component: _index.Form,
  argTypes: {}
};
exports.default = _default;

var FormControlTemplate = function FormControlTemplate(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Form, null, /*#__PURE__*/_react.default.createElement("h3", null, "Select"), /*#__PURE__*/_react.default.createElement(_index.Form.Select, {
    "aria-label": "Default select example"
  }, /*#__PURE__*/_react.default.createElement("option", null, "Open this select menu"), /*#__PURE__*/_react.default.createElement("option", {
    value: "1"
  }, "One"), /*#__PURE__*/_react.default.createElement("option", {
    value: "2"
  }, "Two"), /*#__PURE__*/_react.default.createElement("option", {
    value: "3"
  }, "Three")), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Select size"), /*#__PURE__*/_react.default.createElement(_index.Form.Select, {
    size: "lg"
  }, /*#__PURE__*/_react.default.createElement("option", null, "Large select")), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_index.Form.Select, null, /*#__PURE__*/_react.default.createElement("option", null, "Default select")), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_index.Form.Select, {
    size: "sm"
  }, /*#__PURE__*/_react.default.createElement("option", null, "Small select")));
};

var Select = FormControlTemplate.bind({});
exports.Select = Select;
Select.args = {};
