// react-bootstrap-storybook/dist/Form/Form.Check.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Check = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Form/Check',
  component: _index.Form,
  argTypes: {}
};
exports.default = _default;
var types = ['checkbox', 'radio', 'switch'];

var FormControlTemplate = function FormControlTemplate(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Form, null, /*#__PURE__*/_react.default.createElement("h3", null, "Form.Check Default"), types.map(function (type) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: "default-".concat(type),
      className: "mb-3"
    }, /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
      type: type,
      id: "default-".concat(type),
      label: "default ".concat(type)
    }), /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
      disabled: true,
      type: type,
      label: "disabled ".concat(type),
      id: "disabled-default-".concat(type)
    }));
  }), /*#__PURE__*/_react.default.createElement("h3", null, "Form.Check Inline"), types.map(function (type) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: "inline-".concat(type),
      className: "mb-3"
    }, /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
      inline: true,
      label: "1",
      name: "group1",
      type: type,
      id: "inline-".concat(type, "-1")
    }), /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
      inline: true,
      label: "2",
      name: "group1",
      type: type,
      id: "inline-".concat(type, "-2")
    }), /*#__PURE__*/_react.default.createElement(_index.Form.Check, {
      inline: true,
      disabled: true,
      label: "3 (disabled)",
      type: type,
      id: "inline-".concat(type, "-3")
    }));
  }));
};

var Check = FormControlTemplate.bind({});
exports.Check = Check;
Check.args = {};
