// react-bootstrap-storybook/dist/Accordion/Accordion.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Accordion = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Accordion',
  component: _index.Accordion,
  argTypes: {}
};
exports.default = _default;
var options = [{
  id: '1',
  title: 'Accordion Item #1',
  component: /*#__PURE__*/_react.default.createElement("div", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. 1")
}, {
  id: '2',
  title: 'Accordion Item #2',
  component: /*#__PURE__*/_react.default.createElement("div", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. 2")
}];

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Accordion, args);
};

var _Accordion = Template.bind({});

exports._Accordion = _Accordion;
_Accordion.args = {
  options: options,
  defaultActiveKey: '1'
};
