// react-bootstrap-storybook/dist/Navbars/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Navbars = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

var _excluded = ["data"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var navbarsBgs = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  DARK: 'dark',
  LIGHT: 'light'
};

var Navbars = function Navbars(_ref) {
  var data = _ref.data,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Navbar, props, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Container, null, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Navbar.Brand, {
    href: "#home"
  }, data.title), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Navbar.Toggle, {
    "aria-controls": "basic-navbar-nav"
  }), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Navbar.Collapse, {
    id: "basic-navbar-nav"
  }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Nav, {
    className: "me-auto"
  }, data.leftMenu.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
      key: "key-".concat(index)
    }, item);
  }))), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Navbar.Collapse, {
    className: "justify-content-end"
  }, data.rightMenu.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
      key: "key-".concat(index)
    }, item);
  }))));
};

exports.Navbars = Navbars;
Navbars.defaultProps = {
  data: {},
  bg: 'light',
  expand: "md"
};
Navbars.propTypes = {
  data: _propTypes.default.instanceOf(Object),
  bg: _propTypes.default.oneOf(Object.values(navbarsBgs)),
  expand: _propTypes.default.oneOf([true, 'sm', 'md', 'lg', 'xl', 'xxl'])
};
var _default = Navbars;
exports.default = _default;
