// react-bootstrap-storybook/dist/Navbars/Navbars.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Navbars = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Navbars',
  component: _index.Navbars,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Navbars, args);
};

var _Navbars = Template.bind({});

exports._Navbars = _Navbars;
_Navbars.args = {
  data: {
    title: 'React-Bootstrap',
    leftMenu: [/*#__PURE__*/_react.default.createElement(_index.Nav.Link, {
      href: "#home"
    }, "Home"), /*#__PURE__*/_react.default.createElement(_index.Nav.Link, {
      href: "#link"
    }, "Link"), /*#__PURE__*/_react.default.createElement(_index.NavDropdown, {
      title: "Dropdown",
      id: "basic-nav-dropdown"
    }, /*#__PURE__*/_react.default.createElement(_index.NavDropdown.Item, {
      href: "#action/3.1"
    }, "Action"), /*#__PURE__*/_react.default.createElement(_index.NavDropdown.Item, {
      href: "#action/3.2"
    }, "Another action"), /*#__PURE__*/_react.default.createElement(_index.NavDropdown.Item, {
      href: "#action/3.3"
    }, "Something"), /*#__PURE__*/_react.default.createElement(_index.NavDropdown.Divider, null), /*#__PURE__*/_react.default.createElement(_index.NavDropdown.Item, {
      href: "#action/3.4"
    }, "Separated link"))],
    rightMenu: [/*#__PURE__*/_react.default.createElement(_index.Form, {
      className: "d-flex",
      style: {
        paddingRight: 10
      }
    }, /*#__PURE__*/_react.default.createElement(_index.InputSearch, {
      className: "mr-2",
      size: "sm"
    })), /*#__PURE__*/_react.default.createElement(_index.Dropdown, null, /*#__PURE__*/_react.default.createElement(_index.Dropdown.Toggle, {
      size: "sm",
      variant: "light",
      id: "dropdown-basic"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: "bi bi-person"
    }), " User Name"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Menu, null, /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
      href: "#/action-1"
    }, "Profile"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
      href: "#/action-2"
    }, "Another action"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Divider, null), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
      href: "#/action-3"
    }, "Logout")))]
  }
};
