// react-bootstrap-storybook/dist/Tooltip/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tooltipPlacements = exports.Tooltip = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

require("./index.css");

var _excluded = ["children", "title", "containerClassName"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var tooltipPlacements = {
  AUTO_START: 'auto-start',
  AUTO: 'auto',
  AUTO_END: 'auto-end',
  TOP_START: 'top-start',
  TOP: 'top',
  TOP_END: 'top-end',
  RIGHT_START: 'right-start',
  RIGHT: 'right',
  RIGHT_END: 'right-end',
  BOTTOM_END: 'bottom-end',
  BOTTOM: 'bottom',
  BOTTOM_START: 'bottom-start',
  LEFT_END: 'left-end',
  LEFT: 'left',
  LEFT_START: 'left-start'
};
exports.tooltipPlacements = tooltipPlacements;

var Tooltip = function Tooltip(_ref) {
  var children = _ref.children,
      title = _ref.title,
      containerClassName = _ref.containerClassName,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.OverlayTrigger, _extends({
    overlay: /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tooltip, null, title || children)
  }, props), /*#__PURE__*/_react.default.createElement("div", {
    className: containerClassName
  }, children));
};

exports.Tooltip = Tooltip;
Tooltip.defaultProps = {
  children: '',
  title: '',
  flip: true,
  placement: undefined,
  containerClassName: 'tooltip-container'
};
Tooltip.propTypes = {
  children: _propTypes.default.any,
  title: _propTypes.default.any,
  flip: _propTypes.default.bool,
  placement: _propTypes.default.oneOf(Object.values(tooltipPlacements)),
  containerClassName: _propTypes.default.string
};
var _default = Tooltip;
exports.default = _default;
