// react-bootstrap-storybook/dist/Spinner/Spinner.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Spinner = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Spinner',
  component: _index.Spinner,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("h3", null, "Border Spinner - border"), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Grow Spinner - grow"), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Variants"), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "primary"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "secondary"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "success"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "danger"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "warning"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "info"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "light"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    variant: "dark"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "primary"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "secondary"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "success"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "danger"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "warning"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "info"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "light"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    variant: "dark"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Sizing"), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border",
    size: "sm"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "border"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow",
    size: "sm"
  }), /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    animation: "grow"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Buttons"), /*#__PURE__*/_react.default.createElement(_index.Button, {
    variant: "primary",
    disabled: true
  }, /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    as: "span",
    animation: "border",
    size: "sm",
    role: "status",
    "aria-hidden": "true"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "visually-hidden"
  }, "Loading...")), ' ', /*#__PURE__*/_react.default.createElement(_index.Button, {
    variant: "primary",
    disabled: true
  }, /*#__PURE__*/_react.default.createElement(_index.Spinner, {
    as: "span",
    animation: "grow",
    size: "sm",
    role: "status",
    "aria-hidden": "true"
  }), "Loading..."), /*#__PURE__*/_react.default.createElement("br", null));
};

var _Spinner = Template.bind({});

exports._Spinner = _Spinner;
_Spinner.args = {};
