// react-bootstrap-storybook/dist/InputGroup/InputGroups.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Inputgroup = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Inputgroup',
  component: _index.InputGroup,
  argTypes: {}
};
exports.default = _default;

var Inputgroup = function Inputgroup() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("h3", null, "InputGroup"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "basic-addon1"
  }, "@"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    placeholder: "Username",
    "aria-label": "Username",
    "aria-describedby": "basic-addon1"
  })), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    placeholder: "Recipient's username",
    "aria-label": "Recipient's username",
    "aria-describedby": "basic-addon2"
  }), /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "basic-addon2"
  }, "@example.com")), /*#__PURE__*/_react.default.createElement(_index.Form.Label, {
    htmlFor: "basic-url"
  }, "Your vanity URL"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "basic-addon3"
  }, "https://example.com/users/"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    id: "basic-url",
    "aria-describedby": "basic-addon3"
  })), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, null, "$"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Amount (to the nearest dollar)"
  }), /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, null, ".00")), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, null, "With textarea"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    as: "textarea",
    "aria-label": "With textarea"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Sizing"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    size: "sm",
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "inputGroup-sizing-sm"
  }, "Small"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Small",
    "aria-describedby": "inputGroup-sizing-sm"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "inputGroup-sizing-default"
  }, "Default"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Default",
    "aria-describedby": "inputGroup-sizing-default"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    size: "lg"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Text, {
    id: "inputGroup-sizing-lg"
  }, "Large"), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Large",
    "aria-describedby": "inputGroup-sizing-sm"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Checkboxes and radios"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Checkbox, {
    "aria-label": "Checkbox for following text input"
  }), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Text input with checkbox"
  })), /*#__PURE__*/_react.default.createElement(_index.InputGroup, null, /*#__PURE__*/_react.default.createElement(_index.InputGroup.Radio, {
    "aria-label": "Radio button for following text input"
  }), /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Text input with radio button"
  })), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Button addons"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    placeholder: "Recipient's username",
    "aria-label": "Recipient's username",
    "aria-describedby": "basic-addon2"
  }), /*#__PURE__*/_react.default.createElement(_index.Button, {
    variant: "outline-secondary",
    id: "button-addon2"
  }, "Button")), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Buttons with Dropdowns"), /*#__PURE__*/_react.default.createElement(_index.InputGroup, {
    className: "mb-3"
  }, /*#__PURE__*/_react.default.createElement(_index.FormControl, {
    "aria-label": "Text input with dropdown button"
  }), /*#__PURE__*/_react.default.createElement(_index.DropdownButton, {
    variant: "outline-secondary",
    title: "Dropdown",
    id: "input-group-dropdown-2",
    align: "end"
  }, /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
    href: "#"
  }, "Action"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
    href: "#"
  }, "Another action"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
    href: "#"
  }, "Something else here"), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Divider, null), /*#__PURE__*/_react.default.createElement(_index.Dropdown.Item, {
    href: "#"
  }, "Separated link"))));
};

exports.Inputgroup = Inputgroup;
