// react-bootstrap-storybook/dist/Alert/Alert.stories.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dismissingsuccess = exports.Dismissingerror = exports.Additionalcontent = exports.Links = exports.Examples = exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _index = require("../index");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var _default = {
  title: 'components/Alert',
  component: _index.Alert,
  argTypes: {}
};
exports.default = _default;
var types = ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'light', 'dark'];

var Examples = function Examples() {
  return types.map(function (variant, idx) {
    return /*#__PURE__*/_react.default.createElement(_index.Alert, {
      key: idx,
      variant: variant
    }, "This is a ", variant, " alert\u2014check it out!");
  });
};

exports.Examples = Examples;

var Links = function Links() {
  return types.map(function (variant, idx) {
    return /*#__PURE__*/_react.default.createElement(_index.Alert, {
      key: idx,
      variant: variant
    }, "This is a ", variant, " alert with", ' ', /*#__PURE__*/_react.default.createElement(_index.Alert.Link, {
      href: "#"
    }, "an example link"), ". Give it a click if you like.");
  });
};

exports.Links = Links;

var Additionalcontent = function Additionalcontent() {
  return /*#__PURE__*/_react.default.createElement(_index.Alert, {
    variant: "success"
  }, /*#__PURE__*/_react.default.createElement(_index.Alert.Heading, null, "Hey, nice to see you"), /*#__PURE__*/_react.default.createElement("p", null, "Aww yeah, you successfully read this important alert message. This example text is going to run a bit longer so that you can see how spacing within an alert works with this kind of content."), /*#__PURE__*/_react.default.createElement("hr", null), /*#__PURE__*/_react.default.createElement("p", {
    className: "mb-0"
  }, "Whenever you need to, be sure to use margin utilities to keep things nice and tidy."));
};

exports.Additionalcontent = Additionalcontent;

var Dismissingerror = function Dismissingerror() {
  var _useState = (0, _react.useState)(true),
      _useState2 = _slicedToArray(_useState, 2),
      show = _useState2[0],
      setShow = _useState2[1];

  if (show) {
    return /*#__PURE__*/_react.default.createElement(_index.Alert, {
      variant: "danger",
      onClose: function onClose() {
        return setShow(false);
      },
      dismissible: true
    }, /*#__PURE__*/_react.default.createElement(_index.Alert.Heading, null, "Oh snap! You got an error!"), /*#__PURE__*/_react.default.createElement("p", null, "Change this and that and try again. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Cras mattis consectetur purus sit amet fermentum."));
  }

  return /*#__PURE__*/_react.default.createElement(_index.Button, {
    onClick: function onClick() {
      return setShow(true);
    }
  }, "Show Alert");
};

exports.Dismissingerror = Dismissingerror;

var Dismissingsuccess = function Dismissingsuccess() {
  var _useState3 = (0, _react.useState)(true),
      _useState4 = _slicedToArray(_useState3, 2),
      show = _useState4[0],
      setShow = _useState4[1];

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.Alert, {
    show: show,
    variant: "success"
  }, /*#__PURE__*/_react.default.createElement(_index.Alert.Heading, null, "How's it going?!"), /*#__PURE__*/_react.default.createElement("p", null, "Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Cras mattis consectetur purus sit amet fermentum."), /*#__PURE__*/_react.default.createElement("hr", null), /*#__PURE__*/_react.default.createElement("div", {
    className: "d-flex justify-content-end"
  }, /*#__PURE__*/_react.default.createElement(_index.Button, {
    onClick: function onClick() {
      return setShow(false);
    },
    variant: "outline-success"
  }, "Close me y'all!"))), !show && /*#__PURE__*/_react.default.createElement(_index.Button, {
    onClick: function onClick() {
      return setShow(true);
    }
  }, "Show Alert"));
};

exports.Dismissingsuccess = Dismissingsuccess;
