// react-bootstrap-storybook/dist/Canvas/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Canvas = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

require("./index.css");

var _excluded = ["title", "children"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Header = _reactBootstrap.Offcanvas.Header,
    Title = _reactBootstrap.Offcanvas.Title,
    Body = _reactBootstrap.Offcanvas.Body;

var Canvas = function Canvas(_ref) {
  var title = _ref.title,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Offcanvas, props, /*#__PURE__*/_react.default.createElement(Header, {
    closeButton: true
  }, /*#__PURE__*/_react.default.createElement(Title, null, title)), /*#__PURE__*/_react.default.createElement(Body, null, children));
};

exports.Canvas = Canvas;
Canvas.defaultProps = {
  show: false,
  title: '',
  children: '',
  placement: 'start',
  backdrop: true,
  scroll: false,
  backdropClassName: 'canvas-backdrop',
  onHide: function onHide() {}
};
Canvas.propTypes = {
  show: _propTypes.default.bool,
  title: _propTypes.default.string,
  children: _propTypes.default.any,
  placement: _propTypes.default.oneOf(['start', 'end', 'top', 'bottom']),
  backdrop: _propTypes.default.bool,
  scroll: _propTypes.default.bool,
  backdropClassName: _propTypes.default.string,
  onHide: _propTypes.default.func
};
var _default = Canvas;
exports.default = _default;
