// react-bootstrap-storybook/dist/Tabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tabs = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

var _excluded = ["tabs", "vertical"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Tabs = function Tabs(_ref) {
  var tabs = _ref.tabs,
      vertical = _ref.vertical,
      props = _objectWithoutProperties(_ref, _excluded);

  return vertical ? /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tab.Container, props, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Row, null, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Col, {
    sm: 3
  }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Nav, {
    variant: "pills",
    className: "flex-column"
  }, tabs.map(function (item) {
    return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Nav.Item, {
      key: item.eventKey
    }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Nav.Link, {
      eventKey: item.eventKey
    }, item.title));
  }))), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Col, {
    sm: 9
  }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tab.Content, null, tabs.map(function (item) {
    return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tab.Pane, {
      key: item.eventKey,
      eventKey: item.eventKey
    }, item.component);
  }))))) : /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tabs, props, tabs.map(function (item) {
    return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Tab, {
      key: item.eventKey,
      eventKey: item.eventKey,
      title: item.title,
      disabled: item.disabled
    }, item.component);
  }));
};

exports.Tabs = Tabs;
Tabs.defaultProps = {
  tabs: [],
  className: 'mb-3',
  defaultActiveKey: '',
  vertical: false
};
Tabs.propTypes = {
  tabs: _propTypes.default.arrayOf(Object),
  className: _propTypes.default.string,
  defaultActiveKey: _propTypes.default.string,
  vertical: _propTypes.default.bool
};
var _default = Tabs;
exports.default = _default;
