// react-bootstrap-storybook/dist/Tabs/Tabs.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Tabs = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Tabs',
  component: _index.Tabs,
  argTypes: {}
};
exports.default = _default;
var tabs = [{
  eventKey: 'tab1',
  title: 'Tab 1',
  component: /*#__PURE__*/_react.default.createElement("div", null, "tab 1")
}, {
  eventKey: 'tab2',
  title: 'Tab 2',
  component: /*#__PURE__*/_react.default.createElement("div", null, "tab 2")
}, {
  eventKey: 'tab3',
  title: 'Tab 3',
  component: /*#__PURE__*/_react.default.createElement("div", null, "tab 3"),
  disabled: true
}];

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Tabs, args);
};

var _Tabs = Template.bind({});

exports._Tabs = _Tabs;
_Tabs.args = {
  tabs: tabs,
  defaultActiveKey: 'tab1',
  vertical: false
};
