// react-bootstrap-storybook/dist/Modal/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

require("./index.css");

var _excluded = ["title", "children", "footer", "show", "onHide", "onSave", "onCancel", "backdrop"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var modalSize = {
  SM: 'sm',
  LG: 'lg',
  XL: 'xl'
};

var Modal = function Modal(_ref) {
  var title = _ref.title,
      children = _ref.children,
      footer = _ref.footer,
      show = _ref.show,
      onHide = _ref.onHide,
      onSave = _ref.onSave,
      onCancel = _ref.onCancel,
      backdrop = _ref.backdrop,
      props = _objectWithoutProperties(_ref, _excluded);

  var footerComponent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Button, {
    className: "button-cancel",
    variant: "secondary",
    size: "sm",
    onClick: onCancel
  }, "Cancel"), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Button, {
    className: "button-save",
    variant: "primary",
    size: "sm",
    onClick: onSave
  }, "Save"));

  if (footer || footer === false) {
    footerComponent = footer;
  }

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal, _extends({
    show: show,
    onHide: onHide,
    backdrop: backdrop || 'static',
    keyboard: backdrop === true
  }, props), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal.Header, {
    closeButton: true
  }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal.Title, null, title)), /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal.Body, null, children), footerComponent && /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal.Footer, null, footerComponent));
};

exports.Modal = Modal;
Modal.defaultProps = {
  title: '',
  children: '',
  footer: '',
  show: false,
  backdrop: true,
  size: 'lg',
  centered: true,
  fullscreen: false,
  scrollable: false,
  dialogClassName: 'bootstrap-dialog',
  contentClassName: 'bootstrap-dialog-content',
  backdropClassName: 'bootstrap-dialog-backdrop',
  onHide: undefined,
  onSave: undefined,
  onCancel: undefined
};
Modal.propTypes = {
  title: _propTypes.default.any,
  children: _propTypes.default.any,
  footer: _propTypes.default.any,
  show: _propTypes.default.bool,
  backdrop: _propTypes.default.bool,
  size: _propTypes.default.oneOf(Object.values(modalSize)),
  centered: _propTypes.default.bool,
  fullscreen: _propTypes.default.bool,
  scrollable: _propTypes.default.bool,
  dialogClassName: _propTypes.default.string,
  contentClassName: _propTypes.default.string,
  backdropClassName: _propTypes.default.string,
  onHide: _propTypes.default.func,
  onSave: _propTypes.default.func,
  onCancel: _propTypes.default.func
};
var _default = Modal;
exports.default = _default;
