// react-bootstrap-storybook/dist/Breadcrumb/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Breadcrumb = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

var _excluded = ["data", "onClick"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Item = _reactBootstrap.Breadcrumb.Item;

var Breadcrumb = function Breadcrumb(_ref) {
  var data = _ref.data,
      onClick = _ref.onClick,
      props = _objectWithoutProperties(_ref, _excluded);

  var size = data.length - 1;

  var handleOnClick = function handleOnClick(item) {
    return function (e) {
      onClick(item, e);
    };
  };

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Breadcrumb, props, data.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement(Item, {
      key: "id-".concat(index),
      href: item.href,
      active: index === size,
      onClick: handleOnClick(item)
    }, item.title);
  }));
};

exports.Breadcrumb = Breadcrumb;
Breadcrumb.defaultProps = {
  data: [],
  onClick: function onClick() {}
};
Breadcrumb.propTypes = {
  data: _propTypes.default.arrayOf(Object),
  onClick: _propTypes.default.func
};
var _default = Breadcrumb;
exports.default = _default;
