// react-bootstrap-storybook/dist/Popovers/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Popovers = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

require("./index.css");

var _excluded = ["children", "title", "component", "containerClassName", "backdrop", "onHide", "trigger"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var popoversPlacements = {
  AUTO_START: 'auto-start',
  AUTO: 'auto',
  AUTO_END: 'auto-end',
  TOP_START: 'top-start',
  TOP: 'top',
  TOP_END: 'top-end',
  RIGHT_START: 'right-start',
  RIGHT: 'right',
  RIGHT_END: 'right-end',
  BOTTOM_END: 'bottom-end',
  BOTTOM: 'bottom',
  BOTTOM_START: 'bottom-start',
  LEFT_END: 'left-end',
  LEFT: 'left',
  LEFT_START: 'left-start'
};

var Popovers = function Popovers(_ref) {
  var children = _ref.children,
      title = _ref.title,
      component = _ref.component,
      containerClassName = _ref.containerClassName,
      propBackdrop = _ref.backdrop,
      _onHide = _ref.onHide,
      trigger = _ref.trigger,
      props = _objectWithoutProperties(_ref, _excluded);

  var _React$useState = _react.default.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var backdrop = trigger === 'click' && propBackdrop;

  var popover = /*#__PURE__*/_react.default.createElement(_reactBootstrap.Popover, null, title && /*#__PURE__*/_react.default.createElement(_reactBootstrap.Popover.Header, {
    as: "h3"
  }, title), component && /*#__PURE__*/_react.default.createElement(_reactBootstrap.Popover.Body, null, component));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactBootstrap.OverlayTrigger, _extends({
    show: backdrop ? open : undefined,
    overlay: popover,
    onHide: _onHide,
    trigger: trigger
  }, props), /*#__PURE__*/_react.default.createElement("div", {
    className: containerClassName,
    onClick: function onClick() {
      return backdrop && setOpen(!open);
    }
  }, children)), backdrop && /*#__PURE__*/_react.default.createElement(_reactBootstrap.Modal, {
    show: open,
    onHide: function onHide() {
      setOpen(false);

      _onHide();
    },
    animation: false,
    dialogClassName: "popovers-dialog",
    backdropClassName: "popovers-backdrop"
  }));
};

exports.Popovers = Popovers;
Popovers.defaultProps = {
  children: '',
  title: '',
  component: '',
  flip: true,
  placement: undefined,
  trigger: "click",
  // 'hover', 'click', 'focus'
  containerClassName: 'popovers-container',
  backdrop: true,
  onHide: function onHide() {}
};
Popovers.propTypes = {
  children: _propTypes.default.any,
  title: _propTypes.default.any,
  component: _propTypes.default.any,
  flip: _propTypes.default.bool,
  placement: _propTypes.default.oneOf(Object.values(popoversPlacements)),
  trigger: _propTypes.default.oneOfType([_propTypes.default.oneOf(['hover', 'click', 'focus']), _propTypes.default.instanceOf(Array)]),
  containerClassName: _propTypes.default.string,
  backdrop: _propTypes.default.bool,
  onHide: _propTypes.default.func
};
var _default = Popovers;
exports.default = _default;
