// react-bootstrap-storybook/dist/Popovers/Popovers.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Popovers = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Popovers',
  component: _index.Popovers,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_index.Popovers, args);
};

var _Popovers = Template.bind({});

exports._Popovers = _Popovers;
_Popovers.args = {
  children: /*#__PURE__*/_react.default.createElement(_index.Button, null, "Click me to see"),
  title: 'Popover right',
  component: /*#__PURE__*/_react.default.createElement("div", null, "And here's some ", /*#__PURE__*/_react.default.createElement("strong", null, "amazing"), " content. It's very engaging. right?"),
  flip: true,
  placement: undefined,
  trigger: "click",
  backdrop: true
};
