// react-bootstrap-storybook/dist/ProgressBar/ProgressBar.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Progressbar = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'components/Progressbar',
  component: _index.ProgressBar,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("h3", null, "Default progress bar."), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    now: 60
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "With label"), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    now: 60,
    label: "".concat(60, "%")
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Contextual alternatives"), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    variant: "success",
    now: 40
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    variant: "info",
    now: 20
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    variant: "warning",
    now: 60
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    variant: "danger",
    now: 80
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Striped"), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "success",
    now: 40
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "info",
    now: 20
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "warning",
    now: 60
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "danger",
    now: 80
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Animated"), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    animated: true,
    now: 45
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h3", null, "Stacked"), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, null, /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "success",
    now: 35,
    key: 1
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    variant: "warning",
    now: 20,
    key: 2
  }), /*#__PURE__*/_react.default.createElement(_index.ProgressBar, {
    striped: true,
    variant: "danger",
    now: 10,
    key: 3
  })), /*#__PURE__*/_react.default.createElement("br", null));
};

var Progressbar = Template.bind({});
exports.Progressbar = Progressbar;
Progressbar.args = {};
