// react-bootstrap-storybook/dist/Carousel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Carousel = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

var _excluded = ["data", "maxHeight", "minHeight"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Item = _reactBootstrap.Carousel.Item,
    Caption = _reactBootstrap.Carousel.Caption;

var Carousel = function Carousel(_ref) {
  var data = _ref.data,
      maxHeight = _ref.maxHeight,
      minHeight = _ref.minHeight,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Carousel, props, data.map(function (item) {
    return /*#__PURE__*/_react.default.createElement(Item, {
      style: {
        maxHeight: maxHeight,
        minHeight: minHeight
      },
      key: item.id
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: "d-block w-100",
      src: item.src,
      alt: item.id
    }), /*#__PURE__*/_react.default.createElement(Caption, null, /*#__PURE__*/_react.default.createElement("h3", null, item.title), /*#__PURE__*/_react.default.createElement("p", null, item.subTitle)));
  }));
};

exports.Carousel = Carousel;
Carousel.defaultProps = {
  data: [],
  maxHeight: undefined,
  minHeight: undefined,
  fade: false,
  interval: null
};
Carousel.propTypes = {
  data: _propTypes.default.arrayOf(Object),
  maxHeight: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]),
  minHeight: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]),
  fade: _propTypes.default.bool,
  interval: _propTypes.default.number
};
var _default = Carousel;
exports.default = _default;
