// react-bootstrap-storybook/dist/Badge/Badge.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Badge = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'components/Badge',
  component: _index.Badge,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return Object.values(_index.badgeBgs).map(function (item) {
    return /*#__PURE__*/_react.default.createElement("span", {
      key: item,
      style: {
        padding: 5
      }
    }, /*#__PURE__*/_react.default.createElement(_index.Badge, _extends({
      key: item
    }, args, {
      bg: item
    }), item));
  });
};

var _Badge = Template.bind({});

exports._Badge = _Badge;
