// react-bootstrap-storybook/dist/Badge/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badgeBgs = exports.Badge = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var badgeBgs = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  DARK: 'dark',
  LIGHT: 'light'
};
exports.badgeBgs = badgeBgs;

var Badge = function Badge(_ref) {
  var props = _extends({}, _ref);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Badge, props);
};

exports.Badge = Badge;
Badge.defaultProps = {
  bg: badgeBgs.SECONDARY,
  pill: false,
  text: undefined
};
Badge.propTypes = {
  bg: _propTypes.default.oneOf(Object.values(badgeBgs)),
  pill: _propTypes.default.bool,
  text: _propTypes.default.oneOf(Object.values(badgeBgs))
};
var _default = Badge;
exports.default = _default;
