// react-bootstrap-storybook/dist/Toasts/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Toasts = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

var _excluded = ["header", "children", "position"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Header = _reactBootstrap.Toast.Header,
    Body = _reactBootstrap.Toast.Body;
var toastsBgs = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  DARK: 'dark',
  LIGHT: 'light'
};
var toastsPositions = {
  TOP_START: 'top-start',
  TOP_CENTER: 'top-center',
  TOP_END: 'top-end',
  MIDDLE_START: 'middle-start',
  MIDDLE_CENTER: 'middle-center',
  MIDDLE_END: 'middle-end',
  BOTTOM_START: 'bottom-start',
  BOTTOM_CENTER: 'bottom-center',
  BOTTOM_END: 'bottom-end'
};

var Toasts = function Toasts(_ref) {
  var header = _ref.header,
      children = _ref.children,
      position = _ref.position,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.ToastContainer, {
    className: "p-3",
    position: position
  }, /*#__PURE__*/_react.default.createElement(_reactBootstrap.Toast, props, header && /*#__PURE__*/_react.default.createElement(Header, null, header), children && /*#__PURE__*/_react.default.createElement(Body, null, children)));
};

exports.Toasts = Toasts;
Toasts.defaultProps = {
  header: '',
  children: '',
  show: false,
  position: toastsPositions.TOP_END,
  delay: 3000,
  autohide: false,
  bg: undefined,
  onClose: function onClose() {}
};
Toasts.propTypes = {
  header: _propTypes.default.any,
  children: _propTypes.default.any,
  show: _propTypes.default.bool,
  position: _propTypes.default.oneOf(Object.values(toastsPositions)),
  delay: _propTypes.default.number,
  autohide: _propTypes.default.bool,
  bg: _propTypes.default.oneOf(Object.values(toastsBgs)),
  onClose: _propTypes.default.func
};
var _default = Toasts;
exports.default = _default;
