// react-bootstrap-storybook/dist/Button/Button.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._Button = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'components/Button',
  component: _index.Button,
  argTypes: {}
};
exports.default = _default;

var _Button = function _Button(args) {
  return Object.values(_index.buttonVariant).map(function (variant) {
    return /*#__PURE__*/_react.default.createElement("span", {
      key: variant,
      style: {
        margin: 5
      }
    }, /*#__PURE__*/_react.default.createElement(_index.Button, _extends({}, args, {
      variant: variant
    }), variant));
  });
};

exports._Button = _Button;
