// react-bootstrap-storybook/dist/Button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonSize = exports.buttonVariant = exports.Button = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactBootstrap = require("react-bootstrap");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var buttonVariant = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  DARK: 'dark',
  LIGHT: 'light',
  LINK: 'link',
  OUTLINE_PRIMARY: 'outline-primary',
  OUTLINE_SECONDARY: 'outline-secondary',
  OUTLINE_SUCCESS: 'outline-success',
  OUTLINE_DANGER: 'outline-danger',
  OUTLINE_WARNING: 'outline-warning',
  OUTLINE_INFO: 'outline-info',
  OUTLINE_DARK: 'outline-dark',
  OUTLINE_LIGHT: 'outline-light'
};
exports.buttonVariant = buttonVariant;
var buttonSize = {
  SM: 'sm',
  LG: 'lg'
};
exports.buttonSize = buttonSize;

var Button = function Button(_ref) {
  var props = _extends({}, _ref);

  return /*#__PURE__*/_react.default.createElement(_reactBootstrap.Button, props);
};

exports.Button = Button;
Button.defaultProps = {
  children: '',
  variant: 'secondary',
  size: 'sm',
  active: false,
  disabled: false,
  onClick: undefined
};
Button.propTypes = {
  children: _propTypes.default.any,
  variant: _propTypes.default.oneOf(Object.values(buttonVariant)),
  size: _propTypes.default.oneOf(Object.values(buttonSize)),
  active: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
var _default = Button;
exports.default = _default;
